modules/detection/lidar_driver.py

```diff
@@ -0,0 +1,270 @@
+"""
+Lightware SF45/B Driver
+Lidar documentation: https://support.lightware.co.za/sf45b/#/commands
+NEEDS TO BE TESTED
+"""
+
+import time
+import struct
+import serial
+
+
+class LidarDriver:
+    """
+    Wrapper for lidar
+    """
+
+    def __init__(self, port_name, baudrate, timeout):
+        """
+        port_name: port that the lidar is connected to
+        baudrate: baudrate of the lidar port
+        timeout: timeout for connecting to serial port
+        """
+        self.serial_port = serial.Serial(port_name, baudrate, timeout=timeout)
+        self._packet_parse_state = 0
+        self._packet_payload_size = 0
+        self._packet_size = 0
+        self._packet_data = []
+
+    def _create_crc(self, data):
+        """
+        Create a CRC-16-CCITT 0x1021 hash of the specified data.
+        """
+        crc = 0
+
+        for i in data:
+            code = crc >> 8
+            code ^= int(i)
+            code ^= code >> 4
+            crc = crc << 8
+            crc ^= code
+            code = code << 5
+            crc ^= code
+            code = code << 7
+            crc ^= code
+            crc &= 0xFFFF
+
+        return crc
+
+    def _build_packet(self, command, write, data=None):
+        """
+        Create raw bytes for a packet.
+        """
+        if data is None:
+            data = []
+
+        payload_length = 1 + len(data)
+        flags = (payload_length << 6) | (write & 0x1)
+        packet_bytes = [0xAA, flags & 0xFF, (flags >> 8) & 0xFF, command]
+        packet_bytes.extend(data)
+        crc = self._create_crc(packet_bytes)
```

Comment:
since the only self. we use here is a static helper function, the build_packets method can be static too (if we just remove the create_crc function and put it in here)
up to you whether or not to do it though - separating it is good because it's more readable, combining it allows you to make build_packets static, which is good because it's not supposed to modify anything in the class